File: 2022/01.exs
Code.require_file("../aoc.exs")

defmodule DayOne do
    def solve(elves) do
        elves
        |> Enum.map(fn elf -> Enum.map(String.split(elf, ["\n"]), &String.to_integer/1) end)
        |> Enum.map(&Enum.sum/1)
        |> Enum.sort(&>=/2)
        # part 1
        |> (tap(&(&1 |> hd |> IO.inspect(label: "p1"))))
        # part 2
        |> Enum.take(3)
        |> Enum.sum
        |> IO.inspect(label: "p2")
    end
end

Aoc.readAndSolve("01.input", &DayOne.solve/1, ["\r\n\r\n", "\n\n"])

File: 2022/01.mjs
import {readAndSolve} from '../aoc.mjs'

function solve(lines) {
    const elves = lines.group('', x => +x);
    const calories = elves
        .map(elf => elf.sum())
        .numSort()
        .reverse();

    const p1 = calories[0];
    const p2 = calories.slice(0, 3).sum();
    return {p1, p2};
}

readAndSolve(process.argv[2] || '01.input', solve);